components/MuseumTile.tsx

```diff
@@ -1,39 +1,112 @@
-import {View, Text, ActivityIndicator, StyleSheet} from "react-native";
+import {
+    View,
+    Text,
+    ActivityIndicator,
+    StyleSheet,
+    TouchableOpacity,
+    useColorScheme,
+    TouchableHighlight
+} from "react-native";
 import getMuseum from "../api/getMuseum";
 import FontAwesome from '@expo/vector-icons/FontAwesome';
+import {useNavigation} from "@react-navigation/native";
+import {MainStackNavigation} from "../App";
+import {Image} from 'expo-image';
+import {Appearance} from 'react-native';
+import {MOCK_IMAGE} from "../constans";
+import {
+    darkBackground,
+    darkHighlight,
+    darkSurface,
+    darkText,
+    lightBackground,
+    lightHighlight,
+    lightSurface,
+    lightText
+} from "../colors";
+import useAppColorScheme from "../hooks/useAppColorScheme";
 
 interface MuseumTileProps {
     id: number;
     selected: boolean,
-    onTap: Function,
+    onFavouriteTap: Function,
 }
 
 export default function MuseumTile(props: MuseumTileProps) {
+    const colorScheme = useAppColorScheme();
 
     const museumAction = getMuseum(props.id);
+    const navigation = useNavigation<MainStackNavigation>();
+
+    const onTileTap = () => {
+        if (!museumAction.inProgress) {
+            navigation.navigate("DetailedMuseum", {"museum": museumAction.data!})
+        }
+    }
+
+
+
+    const tileColorScheme = colorScheme.light ? styles.tileLightColorScheme : styles.tileDarkColorScheme;
+    const labelColorScheme = colorScheme.light ? styles.labelLightColorScheme : styles.labelDarkColorScheme;
+
+
+    const underlayColor = colorScheme.light ? lightHighlight : darkHighlight;
+    const iconButtonBackgroundColor = colorScheme.light ? lightSurface : darkSurface;
+    const highlightColor = colorScheme.light ? lightHighlight : darkHighlight;
```

Comment:
If `useAppColorScheme` returns only `"dark" | "light"` and the colors would be nicely structured in an object we could simplify this code here to look like this:

```suggestion
    const underlayColor = Colors[theme].highlight;
    const iconButtonBackgroundColor = Colors[theme].surface;
    const highlightColor = Colors[theme].highlight;
```